presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/FunctionCallProvider.java

```diff
@@ -98,12 +100,28 @@ public boolean equals(Object object)
             FunctionCall other = (FunctionCall) object;
 
             return Objects.equals(name, other.getName()) &&
-                    other.getWindow().isPresent() &&
-                    Objects.equals(frame, other.getWindow().get().getFrame()) &&
+                    matchWindowFrame(other.getWindow()) &&
                     Objects.equals(distinct, other.isDistinct()) &&
                     Objects.equals(getArguments(), other.getArguments());
         }
 
+        private boolean matchWindowFrame(Optional<Window> other)
+        {
+            boolean doesNotHaveOtherFrame = !other.isPresent()
+                    || !(other.get() instanceof WindowInline)
+                    || !(((WindowInline) other.get()).getFrame().isPresent());
+
+            if (!frame.isPresent() && doesNotHaveOtherFrame) {
```

Comment:
My fault, you're comparing frames, not windows here.
So instead of the above, you can do:
```java
Optional<Frame> other = other.filter(WindowInline.class::isInstance).flatMap(WindowInline.class::cast).flatMap(Window::getFrame);
return Objects.equals(frame, other);
```